### PR TITLE
challenge integration tests

### PR DESCRIPTION
**Pull request by syuhei176**
These scenarios don't pass yet.

### CODE REVIEW

File: packages/ovm/src/deciders/operators/ForAllSuchThatDecider.ts [hunk 1]
```diff
@@ -65,7 +81,7 @@ export class ForAllSuchThatDecider implements Decider {
         }
         return {
           outcome: false,
-          witnesses: [],
+          witnesses: undefined,
```
[the remark this answers is not in the document]
Comment (syuhei176) [reply packages/ovm/src/deciders/operators/ForAllSuchThatDecider.ts]: fixed

**Comment by syuhei176** [comment]
I'll update around recoverHint method and add test!

File: packages/ovm/__tests__/decompiler/CompiledPredicate.test.ts [hunk 2]
```diff
@@ -38,6 +38,29 @@ describe('CompiledPredicate', () => {
...
+  Bool(a, b) and Bool(a, b)
+
+def test(a) := SampleQuantifier(a).any(b -> b())
+`
+      )
+      const hint = compiledPredicate.recoverHint(
+        new Property(TestPredicateAddress, [
+          PredicateLabel.from('TestTA'),
+          Coder.encode(BigNumber.from(1)),
+          Coder.encode(BigNumber.from(2))
```
Comment (tkmct) [inline packages/ovm/__tests__/decompiler/CompiledPredicate.test.ts:57]: I'm not sure but shouldn't this second input be range?

File: packages/ovm/src/decompiler/CompiledDecider.ts [hunk 3]
```diff
@@ -3,6 +3,7 @@ import { Decider } from '../interfaces/Decider'
 import { Decision, Property } from '../types'
 import { DeciderManager } from '../DeciderManager'
 import { CompiledPredicate } from './CompiledPredicate'
+import { ThereExistsSuchThatDecider } from '../deciders'
```
Comment (tkmct) [inline packages/ovm/src/decompiler/CompiledDecider.ts:6]: seems not to be used

File: packages/ovm/src/deciders/operators/ForAllSuchThatDecider.ts [hunk 4]
```diff
@@ -18,15 +39,16 @@ export class ForAllSuchThatDecider implements Decider {
...
   ): Promise<Decision> {
-    let witnesses: Bytes[]
-    if (isHint(inputs[0])) {
-      witnesses = await getWitnesses(
-        manager.witnessDb,
-        replaceHint(inputs[0].intoString(), substitutions)
-      )
-    } else {
+    if (inputs[0].equals(Bytes.fromString(''))) {
+      inputs[0] = recoverHint(manager, inputs)
```
Comment (tkmct) [inline packages/ovm/src/deciders/operators/ForAllSuchThatDecider.ts:43]: hmm, I'm not sure if it's good to overwrite into inputs in decider

Comment (syuhei176) [reply packages/ovm/src/deciders/operators/ForAllSuchThatDecider.ts:43]: ah, I think it's not good. I'll fix.

File: packages/ovm/__tests__/decompiler/CompiledPredicate.test.ts (hunk 2 above)
Comment (syuhei176) [reply packages/ovm/__tests__/decompiler/CompiledPredicate.test.ts:57]: It should be range, but this test doesn't evaluate hint string. well, I believe it'ok.

File: packages/ovm/src/decompiler/CompiledDecider.ts (hunk 3 above)
Comment (syuhei176) [reply packages/ovm/src/decompiler/CompiledDecider.ts]: ok!

File: packages/ovm/src/deciders/operators/ForAllSuchThatDecider.ts (hunk 4 above)
Comment (syuhei176) [reply packages/ovm/src/deciders/operators/ForAllSuchThatDecider.ts]: I moved restoring process.

File: packages/ovm/src/deciders/operators/ForAllSuchThatDecider.ts [hunk 5]
```diff
@@ -4,6 +4,7 @@ import { Decider } from '../../interfaces/Decider'
 import { Decision, Property, Challenge, LogicalConnective } from '../../types'
 import { DeciderManager } from '../../DeciderManager'
 import { TraceInfoCreator } from '../../Tracer'
+import { CompiledDecider } from '../../decompiler'
```
Comment (tkmct) [inline packages/ovm/src/deciders/operators/ForAllSuchThatDecider.ts:7]: no longer used?

Comment (syuhei176) [reply packages/ovm/src/deciders/operators/ForAllSuchThatDecider.ts]: sorry, fixed!

Comment (tkmct) [reply packages/ovm/src/deciders/operators/ForAllSuchThatDecider.ts]: 👍

**Pull request merged**